### PR TITLE
init Tekton UI commit

### PR DESCRIPTION
**Pull request by drewbutlerbb4**
Initial commit for Tekton Graph display support in the KFP UI.

- Modified creation of static and runtime execution graphs to support Tekton graphs
- Modified supporting functions which parse descriptive information of each Tekton Task, Condition, and PipelineRun

### CODE REVIEW

**Comment by Tomcli** [comment]
@drewbutlerbb4 can you also include the "step-main" fixes for the UI server logging? Thanks.

File: frontend/server/k8s-helper.ts [hunk 1]
```diff
@@ -214,7 +214,7 @@ export function getPodLogs(podName: string, podNamespace?: string): Promise<stri
       `podNamespace is not specified and cannot get namespace from ${namespaceFilePath}.`,
     );
   }
-  return (k8sV1Client.readNamespacedPodLog(podName, podNamespace, 'main') as any).then(
+  return (k8sV1Client.readNamespacedPodLog(podName, podNamespace, 'step-main') as any).then(
```
Comment (animeshsingh) [inline frontend/server/k8s-helper.ts:217]: items for next iteration - add a comment for why the change was needed?

File: frontend/src/lib/StaticGraphParser.ts [hunk 2]
```diff
@@ -52,214 +48,105 @@ export class SelectedNodeInfo {
   }
 }
 
-export function _populateInfoFromTemplate(
```
Comment (animeshsingh) [inline frontend/src/lib/StaticGraphParser.ts:55]: this function is argo specific - will need to get a generic function to handle both

File: frontend/src/lib/StaticGraphParser.ts [hunk 3]
```diff
@@ -52,214 +48,105 @@ export class SelectedNodeInfo {
...
+export function createGraph(workflow: any): dagre.graphlib.Graph {
+  const graph = new dagre.graphlib.Graph();
+  graph.setGraph({});
+  graph.setDefaultEdgeLabel(() => ({}));
 
-  if (root && root.nodeType === 'dag') {
-    // Mark that we have visited this DAG, and save the original qualified path to it.
-    alreadyVisited.set(rootTemplateId, parentFullPath || '/' + rootTemplateId);
-    const template = root.template;
+  buildTektonDag(graph, workflow);
```
Comment (animeshsingh) [inline frontend/src/lib/StaticGraphParser.ts:82]: come back to see if there can be a generic buildDag() to then call specific implementations based on Tekton or Argo

**Comment by animeshsingh** [comment]
@Tomcli @drewbutlerbb4 any reason why python 3.5 build is failing?
https://github.com/kubeflow/kfp-tekton-backend/pull/5/checks?check_run_id=721748575

**Comment by Tomcli** [comment]
The same test failed occasionally in kubeflow/pipeline as well. Looks like is more a Travis issue where sometime the python environment couldn't install some dependencies such as the Tensorflow package.

**Comment by Tomcli** [comment]
also the retest doesn't work on this repo, we might have to create a new commit to trigger the travis test.

**Comment by animeshsingh** [comment]
hmmm - why does retest doesn't work @Tomcli ?

**Comment by Tomcli** [comment]
I don't have access to the ci bot, so I'm not sure is it a config issue.

**Comment by jlewi** [comment]
I think the chatbot commands e.g. "/test" are prow commands. It looks like you are using travis here so the prow commands won't have any impact on travis.

**Comment by Tomcli** [comment]
@drewbutlerbb4 can you rebase this PR to pass the test? thanks

**Comment by drewbutlerbb4** [comment]
@Tomcli I have rebased it and it looks like all the tests have passed, thanks for the help!
